egs/sre16/v2/get_egs_v2.sh

```diff
@@ -0,0 +1,298 @@
+  for n in `seq $num_train_archives`; do
+    utils/filter_scp.pl ${data_combine}/split/${n}/feats.scp ${data_combine}/utt2spk \
+      > ${data_combine}/split/${n}/utt2spk
+    utils/utt2spk_to_spk2utt.pl ${data_combine}/split/${n}/utt2spk \
+      > ${data_combine}/split/${n}/spk2utt
+    utils/sym2int.pl -f 2 ${data}/spk2int ${data_combine}/split/${n}/utt2spk \
+      > ${data_combine}/split/${n}/utt2int
+  done
+fi
+
+if [ $stage -le 3 ]; then
+  echo "$0: Dump Egs for training set"
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [[ ! -d $dir/storage ]]; then
+    utils/create_split_dir.pl \
+      /export/b{03,04,05,06}/$USER/kaldi-data/egs/sre16/v2/xvector-$(date +'%m_%d_%H_%M')/storage $dir/storage
+  fi
+  if [ -e $dir/storage ]; then
+    echo "$0: creating data links"
+    utils/create_data_link.pl $(for x in $(seq $num_train_archives); do echo $dir/egs.$x.ark; done)
+  fi
+  #Note: if --num-pdfs options is not supply, you must use global utt2int,
+  #because we will compute the num-pdfs from utt2label file.
+  $train_cmd --max-jobs-run $nj JOB=1:$num_train_archives $dir/log/dump_egs.JOB.log \
+    nnet3-xvector-get-egs-seg \
+      --num-pdfs=5139 \
+      $data/train_set/temp/combine/utt2int \
+      scp:$data/train_set/temp/combine/split/JOB/feats.scp \
+      ark,scp:$dir/egs.JOB.ark,$dir/egs.JOB.scp
+fi
+
+if [ $stage -le 4 ]; then
+  echo "$0: Dealing with validation set"
+  #Generate data_li/feat.scp and utt2spk
+  $cmd $dir/log/allocate_examples_validate.log \
+    allocate_egs_seg_v2.py \
+      --min-frames-per-chunk=$min_frames_per_chunk \
+      --max-frames-per-chunk=$max_frames_per_chunk \
+      --kinds-of-length=$kinds_of_length_valid \
+      --generate-validate=true \
+      --randomize-chunk-length=false \
+      --data-dir=$data/valid_set \
+      --output-dir=$data/valid_set/temp || exit 1
+  #Generate data_li/spk2utt and utt2label
+  for subdir in `dir $data/valid_set/temp`; do
+    utils/utt2spk_to_spk2utt.pl $data/valid_set/temp/$subdir/utt2spk > $data/valid_set/temp/$subdir/spk2utt
+    utils/sym2int.pl -f 2 $data/spk2int $data/valid_set/temp/$subdir/utt2spk \
+      > $data/valid_set/temp/$subdir/utt2int
+  done
+
+  echo "$0: combine and shuffle (valid)"
+  #combine feats.scp and shuffle
+  mkdir -p $data/valid_set/temp/combine
+  data_combine=$data/valid_set/temp/combine
+
+  cat $data/valid_set/temp/data_*/feats.scp > ${data_combine}/feats.scp.bak
+  utils/shuffle_list.pl ${data_combine}/feats.scp.bak > ${data_combine}/feats.scp
+  #combine utt2spk
+  cat $data/valid_set/temp/data_*/utt2spk > ${data_combine}/utt2spk
+  #generate spk2utt and utt2label
+  utils/utt2spk_to_spk2utt.pl ${data_combine}/utt2spk > ${data_combine}/spk2utt
+  utils/sym2int.pl -f 2 $data/spk2int ${data_combine}/utt2spk \
+    > ${data_combine}/utt2int
+
+  # split feats.scp
+  directories=$(for n in `seq $kinds_of_length_valid`; do echo ${data_combine}/split/${n}; done)
```

Comment:
Why does the number of splits here need to be equal to `$kinds_of_length_valid`? It is ok to do that, but probably not required.
I think if the number of valid egs is small compared to the training egs, it can be done in the background. See how it is done in `steps/nnet3/chain/get_egs.sh`.